cluster template using an existing vpc and tableau version 2020.2.6

albac opened the pull request:
*Issue #, if available:
Only single node supports existing VPC, we need a cluster template we can use with existing VPC.
Table Version is already on 2020.3, we require to test  version of tableau 2020.2.6 which is not in aws repositories and jdbc driver needs to be tested with this repo. 

*Description of changes:
Added new template for cluster using an existing VPC. Tested using taskcat tableau version 2020.2.6 and latest jdbc drivers.
Also found and fixed some taskcat lints from latest version which are not supporting strings on ports which must be numbers.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

vsnyc commented:
@albac - Alfredo, thank you for creating this.

@kglover29 - Kevin, can you look at this PR? The main thing I want to check is if it is okay to download the Tableau binaries from the official location, e.g. https://downloads.tableau.com/esdalt/2020.2.6/tableau-server-2020-2-6.x86_64.rpm. Not sure if there is a good reason to download it from the tableau-quickstart S3 bucket.

albac commented:
@vsnyc, @kglover29,

I just added a fix for a typo I had on this PR, also I removed the Availability zone since it is not needed for existing subnets, also added the fix for the node roles set I saw this on another PR, I needed this change to make the cluster test work.

In regards to the tableau version I wouldn't mind to use the tableau-quickstart S3 bucket but this version wasn't there and I am not sure what the process is to add a new RPM so I used the one available on the tableau repositories.

albac commented:
Just realized that the existing template supports already an existing vpc, I removed my template from this PR. However I added the test on taskcat.yaml so there is an example on how to run this for an existing vpc. I added a new instance type and added the topology timeout as parameter since I needed to increase the timeout.

vsnyc commented:
Cluster templates are removed in the latest version: https://github.com/aws-quickstart/quickstart-tableau-server/pull/100
